src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,28 @@
 package mate.academy;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final ConcurrentHashMap<EventListener, Boolean> listeners = new ConcurrentHashMap<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.put(listener, true);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners.keySet()) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
```

Comment:
Submitting tasks to an ExecutorService inside a loop can result in out-of-order execution of event notifications if event order is significant. If the order matters, you might need to consider alternatives to ensure that events are processed in the order they are received.